ecdh.go

```diff
@@ -101,7 +99,7 @@ func (k *PrivateKeyECDH) PublicKey() (*PublicKeyECDH, error) {
 			return nil, newOpenSSLError("EVP_PKEY_get_octet_string_param")
 		}
 		bytes = C.GoBytes(unsafe.Pointer(cbytes), C.int(n))
-		C.free(unsafe.Pointer(cbytes))
```

Comment:
Is this necessary for supporting Windows? If not, better do it in another PR.

Reply:
It's not _strictly_ necessary if the C toolchain used with cgo links against the same CRT as the OpenSSL DLL. I'll split this change into a separate PR.

Reply:
The version of MinGW-w64 installed on the Windows Server 2022 GHA runner image links against a different CRT than the preinstalled OpenSSL libraries. Therefore this _is_ necessary to support running Windows CI tests on GHA using the default toolchains.